src/animation/fluidResistance.ts

```diff
@@ -0,0 +1,135 @@
+import { Listener, AnimationParams, AnimationInitializer } from './types';
+
+export interface FluidResistance1DParams extends AnimationParams {
+  config: {
+    mass: number;
+    rho: number;
+    area: number;
+    cDrag: number;
+    settle?: boolean;
+  };
+}
+
+interface FluidResistanceState {
+  mover: Entity;
+}
+
+/**
+ * A function to apply the drag force on each step in the course
+ * of the animation. First, we derive the force vector exerted by the
+ * fluid on the mover. Then we apply that vector to the mover to determine
+ * its next acceleration, velocity, and position.
+ */
+const applyFluidResistanceForceForStep = (
+  { mover }: FluidResistanceState,
+  config: FluidResistance1DParams['config']
+) => {
+  const dragForce = fluidResistanceForceV({
+    rho: config.rho,
+    area: config.area,
+    velocity: mover.velocity,
+    cDrag: config.cDrag,
+  });
+
+  const gravitationalForce: Vector<number> = [0, mover.mass * gE];
+  const force = multf({
+    v: addf({ v1: dragForce, v2: gravitationalForce }),
+    s: 0.001,
+  });
+
+  return applyForce({ force, entity: mover });
+};
+
+/**
+ * The fluid resistance function. This function tracks the internal state of the
+ * mover and starts the frame loop to apply the drag force as it moves.
+ */
+export const fluidResistance1D = ({
+  config,
+  onUpdate,
+  onComplete,
+}: FluidResistance1DParams): { controller: AnimationInitializer } => {
+  const state: FluidResistanceState = {
+    mover: {
+      mass: config.mass,
+      acceleration: [0, 0],
+      velocity: [0, 0],
+      position: [0, 0],
+    },
+  };
+
+  const tvPosition = getFluidPositionAtTerminalVelocity(config);
+
+  const listener: Listener = (timestamp, lastFrame, stop) => {
+    /**
+     * Determine the number of milliseconds elapsed between the current frame
+     * and the last frame. If more than four frames have been dropped, assuming
+     * a 60fps interval, just use the timestamp of the current frame.
+     */
+
+    // Obtain the timestamp of the last frame. If this is the first frame, use the current frame timestamp.
+    let lastTime = lastFrame !== undefined ? lastFrame : timestamp;
+
+    /**
+     * If more than four frames have been dropped since the last frame,
+     * just use the current frame timestamp.
+     */
+    if (timestamp > lastTime + 64) {
+      lastTime = timestamp;
+    }
+
+    // Determine the number of steps between the current frame and last recorded frame.
+    const steps = Math.floor(timestamp - lastTime);
+
+    // Apply the drag force once for each step.
+    for (let i = 0; i < steps; i++) {
+      // If applying a settle effect, reverse and scale down the mover's velocity.
+      if (config.settle && state.mover.position[1] >= tvPosition) {
+        state.mover = {
+          ...state.mover,
+          velocity: multf({ v: state.mover.velocity, s: -0.75 }),
+          position: [0, tvPosition],
+        };
+      }
```

Comment:
Here's how we achieve the settle effect. Basically, we reverse the velocity vector of the mover and scale it down a little bit. We could consider exposing `s` here as a user parameter so they can set how much they want to scale the velocity down by. 0.75 is rather arbitrary here and just represents what "felt good".

Reply:
* Scratch that, since it is in the `applyForce` function it'll need to come up separately, since I don't want to disrupt the other local changes I have here. I'm going to merge as is, and then make a dedicated PR to fixing this across all three forces. Thanks again for your help here!